### ISSUE TITLE
[tuify] API change the return type of `select_from_list()`

### ISSUE THREAD

**Issue by NadiaIdris** [r3bl-org/r3bl-open-core#200]
Note - this is breaking change for existing dependents of this crate. We should have a migration strategy in place for them. Either preserve the existing API and add a new one, instead of this breaking change, or just submit a PR w/ the use of the new API to their repo (eg: ockam).

File: `state.rs`
https://github.com/r3bl-org/r3bl-open-core/blob/main/tuify/src/state.rs#L23-L49

1. Change `selected_items` type to be `SelectModeResult`. Please see below.
2. `select_from_list` function should return a `SelectModeResult` and not `Option<Vec<String>>`.
3. Write tests.

```rust
pub struct State {
    /// Does not include the header row.
    pub max_display_height: ChUnit,
    pub max_display_width: ChUnit,
    /// This is not adjusted for
    /// [scroll_offset_row_index](State::scroll_offset_row_index).
    pub raw_caret_row_index: ChUnit,
    pub scroll_offset_row_index: ChUnit,
    pub items: Vec<String>,
    pub selected_items: SelectModeResult,
    pub header: String,
    pub selection_mode: SelectionMode,
}

#[derive(Debug, PartialEq, Eq, Hash, Clone, Default)]
enum SelectModeResult {
    Single(String),
    Multiple(Vec<String>),
    #[default]
    None,
}
```

Also add support to detect <kdb>Ctrl+C</kbd>. The behavior of pressing <kbd>Ctrl+C</kbd> is the same as pressing <kbd>Esc</kbd>.
- https://github.com/r3bl-org/r3bl-open-core/blob/main/tuify/src/keypress.rs#L29
- https://github.com/r3bl-org/r3bl-open-core/blob/main/tuify/src/keypress.rs#L50
- https://github.com/r3bl-org/r3bl-open-core/blob/main/tuify/src/keypress.rs#L50
- Need to use crossterm to detect <kdb>Ctrl+C</kbd>: https://github.com/crossterm-rs/crossterm/issues/655

**Comment by NadiaIdris** [comment]
@Harshil-Jani I removed both of us from this issue for the time being (since we weren't working on this issue immediately) because I wanted to post it to [Rust user forum](https://users.rust-lang.org/) to invite contributors to work on some issues. But my account is still on hold with them and I wasn't able to post them 😀. Trying to get my account "unblocked".

**Comment by nazmulidris** [comment]
@NadiaIdris Should we also add "Ctrl_C_Pressed" as a variant for the `SelectModeResult` enum?

```rust
enum SelectModeResult {
    Single(String),
    Multiple(Vec<String>),
    None,
    Ctrl_C_Pressed,
}
```

Also this would require adding support for trapping <kbd>ctrl+c</kbd> in the main event loop as well.

**Comment by NadiaIdris** [comment]
@nazmulidris What's the difference between `None` and `Ctrl_C_Pressed`? Currently if I am selecting something from single select component and press `Esc` it displays a text "user selected nothing".

**Comment by nazmulidris** [comment]
You are right. There is no need for a Ctrl+C variant. Then all that needs to be added to main event loop is ctrl + c detection which should behave the same way as pressing esc.

**Comment by zoo868e** [comment]
Cool, can I give it a shot?

**Comment by zoo868e** [comment]
Just a quick question: Why is the default value an empty string instead of None?

**Comment by NadiaIdris** [comment]
> Cool, can I give it a shot?

Great! I have assigned you to the issue. Let us know if you need any help on Discord https://discord.com/invite/rkBsNK83.

**Comment by NadiaIdris** [comment]
> Just a quick question: Why is the default value an empty string instead of None?

@zoo868e You are correct. The default should be None 👍🏼
